contrib/opencensus-ext-azure/opencensus/ext/azure/common/__init__.py

```diff
@@ -39,3 +40,21 @@ class Options(BaseObject):
         storage_retention_period=7*24*60*60,
         timeout=10.0,  # networking timeout in seconds
     )
+
+
+def validate_key(instrumentation_key):
+    if not instrumentation_key:
+        raise ValueError("Instrumentation key cannot be none or empty.")
+    # Validate UUID format
+    # Specs taken from https://tools.ietf.org/html/rfc4122
+    pattern = re.compile('[0-9a-f]{8}-'
```

Comment:
This precompiled regular expression should be created once, not inside the function.